src/PetriNet/PNSolver.hs

```diff
@@ -680,14 +681,15 @@ findProgram env dst st ps
         disableDemand <- getExperiment disableDemand
         disableRele <- getExperiment disableRelevancy
         params <- gets $ view searchParams
-        checkedSols <-
-            withTime
-                TypeCheckTime
-                (filterM (liftIO . runGhcChecks params env dst) [code'])
-        if (code' `elem` solutions) || null checkedSols
+        fState <- gets $ view filterState
+        (checkResult, fState') <-
+            withTime TypeCheckTime (liftIO $ runStateT (runGhcChecks params env dst code') fState)
+        modify $ set filterState fState'
```

Comment:
This is what I had in mind, thanks for threading state like this.

That said, we throw around words like `check`, `filter`, `result` quite a bit in this feature. Slightly different names and phrasing variables to reflect their type _really_ helps readability. So in this case, since it's a boolean, the variable should read as an implicit question (`isX` or `areY` are common patterns across many languages and styles)